_ext/asciidoctor_toc.rb

```diff
@@ -0,0 +1,18 @@
+require 'asciidoctor'
+
+module Awestruct::Extensions
+  class AsciidoctorTOC
+    def execute(site)
+      site.pages.each do |page|
+        if page.source_path.end_with? '.adoc'
+          document = Asciidoctor.load_file page.source_path, header_footer: true, safe: :safe, attributes: 'idprefix idseparator=-'
+          page.toc = true
```

Comment:
Yes, I think the right solution is to put the Asciidoctor::Document into the page context (perhaps enabled globally via an option in the site configuration). I'd still be open to using an extension that provides a simpler API to generate the toc (hiding the call to the outline handler), but we shouldn't have to parse the document a second time. That would be nice to avoid.